doc/development/deprecations.rst

```diff
@@ -9,6 +9,11 @@ deprecations are listed below.
 Pending deprecations
 --------------------
 
+* The ``simplify`` argument in ``qml.ops.Hamiltonian`` and ``qml.ops.LinearCombination`` is deprecated. 
+  Instead, ``qml.simplify()`` can be called on the constructed operator.
+
+  - Deprecated in v0.37
```

Comment:
Let's set the target release as `v0.39` since `Hamiltonian` is more widely used :+1: